backtrace-library/src/androidTest/java/backtraceio/library/BacktraceClientMetricsTest.java

```diff
@@ -0,0 +1,938 @@
+package backtraceio.library;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static junit.framework.TestCase.fail;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static java.lang.Thread.sleep;
+
+import android.content.Context;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.runner.AndroidJUnit4;
+
+import net.jodah.concurrentunit.Waiter;
+
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import backtraceio.library.common.BacktraceSerializeHelper;
+import backtraceio.library.common.DeviceAttributesHelper;
+import backtraceio.library.logger.BacktraceLogger;
+import backtraceio.library.logger.LogLevel;
+import backtraceio.library.metrics.BacktraceMetrics;
+import backtraceio.library.metrics.Event;
+import backtraceio.library.metrics.EventsOnServerResponseEventListener;
+import backtraceio.library.metrics.EventsPayload;
+import backtraceio.library.metrics.EventsRequestHandler;
+import backtraceio.library.metrics.EventsResult;
+import backtraceio.library.metrics.UniqueEvent;
+import backtraceio.library.models.BacktraceMetricsSettings;
+import backtraceio.library.models.json.BacktraceAttributes;
+import backtraceio.library.models.types.BacktraceResultStatus;
+
+@RunWith(AndroidJUnit4.class)
+public class BacktraceClientMetricsTest {
```

Comment:
Recommendation: 
* right now file has ~700lines with mixed logic of unique and summed events - wouldn't it be better to keep our tests simple and instead of mix everything in one file, have separated tests for unique and summed events? Right now my browser refuse to work with that huge file ^^

Reply:
Will do!